### PR TITLE
Adding NavItem

### LINKED ISSUE #5911: Create `NavItem` Component
Create a `NavItem` component using the new [figma design (direct link to Nav)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=422-9155&mode=dev) and [here is Footer](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=696-9074&mode=dev). 

This component is a link with additional styling for focus states, and an indicator if the href is an external link. This one component will be used for both the `Navbar` (green accents) and the `Footer` (grey accents) in future issues.

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).
- Review the [component creation guide](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-components) to follow our standards.

---

### Details

image

image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Create a new directory called `NavItem` within the [`sections` directory](https://github.com/nodejs/nodejs.org/tree/main/components/sections) that contains all the new code. `sections` may not exist yet
- Use the new tailwind color and typography system to assign properties. For example, the NavItem active states uses `Green/600` per the figma.
- Add a storybook story file `components/sections/NavItem/index.stories.tsx` which exercises each of the component's states.
- Use intelligent detection of the `href` host compared to the current host to determine if the link is external. If it is, render `arrow-up-right` icon from the hero icons dependency. This only applies for `type` === `nav`

Suggested props include:

- `label` (string): `Learn` in the figma
- `href` (`typeof next/link`)
- `type` (enum, `nav` | `footer`)

There are 8 Navbar states to capture within styles and stories:

- Light, internal link
- Light, internal link, active (focused or hovered)
- Light, external link
- Light, external link, active (focused or hovered)
- Dark, internal link
- Dark, internal link, active (focused or hovered)
- Dark, external link
- Dark, external link, active (focused or hovered)

There are 4 footer states to capture within styles and stories:

- Light
- Light, active (focused or hovered)
- Dark
- Dark, active (focused or hovered)

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
## Description
Issue Solved : Create NavItem Component #5911

created NavItem  component wtih help of typescript and tailwind css whole design is copied from the figma file provided by admin 

## Validation

*Dark Mode*
image

*Light Mode*
image

Also in the active stage the button gets converted in green color as specified in the figma file.

## Related Issues

Create NavItem Component #5911

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by DhairyaMajmudar** [comment]
@ovflowd Bhaiya as instructed I opened a pull request for this issue . Even though I tried my level best to do it on first but failed by few hours 🙏 . I would be blessed if you accept my PR and it will be an honor for me to contribute in the codebase of NodeJS website

File: .idea/modules.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AugustinMauroy) [inline .idea/modules.xml:1]: Should we accept idea IDE in this pr ?

File: components/sections/NavItem/index.module.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
+.navItem {
+  @apply inline-flex items-center gap-2 rounded px-3 py-2;
```
Comment (AugustinMauroy) [inline components/sections/NavItem/index.module.css:2]: Remember to reread the [contribution guideline](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-react-components) 😄
[Example of how it's should write](https://github.com/nodejs/nodejs.org/blob/main/components/Common/Banner/index.module.css)

**Review by AugustinMauroy — COMMENTED** [review]
Good first contribution but some improvement can be done

File: components/sections/NavItem/index.module.css (hunk 2 above)
Comment (DhairyaMajmudar) [reply components/sections/NavItem/index.module.css:2]: Okay Sir 😅

File: .idea/modules.xml (hunk 1 above)
Comment (ovflowd) [reply .idea/modules.xml:1]: No. All these files should be deleted.

**Review by ovflowd — CHANGES_REQUESTED** [review]
I can't imagine how your React Component is a 1:1 match of the other PR that was opened a few hours before yours.

It's impossible that you've made a 1:1 match of their PRs. I'm concluding that this violates our Contribution Guidelines and closing this PR.

**Comment by ovflowd** [comment]
If this is not the case, please feel free to justify your reasoning, and I apologize for any inconvenience.

**Comment by DhairyaMajmudar** [comment]
> If this is not the case, please feel free to justify your reasoning, and I apologize for any inconvenience.

Sir to be honest I have created the components by my own with the help of JavaScript in a dummy folder since I lack knowledge of Typescript I referenced the PR and took code from it

I am really very sorry for the same I hope you understand and forgive me 🙏🏻

**Comment by ovflowd** [comment]
> > If this is not the case, please feel free to justify your reasoning, and I apologize for any inconvenience.
> 
> Sir to be honest I have created the components by my own with the help of JavaScript in a dummy folder since I lack knowledge of Typescript I referenced the PR and took code from it
> 
> I am really very sorry for the same I hope you understand and forgive me 🙏🏻

Hey there. I understand that on your urge to make a contribution, and due to lack of knowledge you thought the best approach here was to go to another PR and copy their codebase.

But that is never the solution. It's a non-ethical approach. Giving credits or not. Both of you want to make a meaningful contribution, and I (we) appreciate that, but there are ways of doing this without putting someone's effort in detriment.

If you lack knowledge or are a novice, we're here to help. PRs can be opened as a draft, and you can always ask for support, guidance or instructions.

There's nothing really to forgive as I doubt you had ill intents; But sadly I cannot allow this PR, hence it will not get merged.

I hope you find this opportunity to find another issue soon (We're opening more issues soon) and be able to still contribute. I appreciate your energy and gesture, but keep in mind that this is not a competition about who merges first or et cetera;

Have a good one!

**Comment by DhairyaMajmudar** [comment]
Thank you so Bhaiya I will always remember the same 🙏🏻 And try my level best to contribute in this prestigious org.

**Pull request closed without merging**